server/pulp/server/db/model/__init__.py

```diff
@@ -315,6 +317,7 @@ class TaskStatus(AutoRetryDocument, ReaperMixin):
     start_time = ISO8601StringField()
     finish_time = ISO8601StringField()
     result = DynamicField()
+    group_id = StringField(default=None)
```

Comment:
If this is switched to a UUID the mongoengine validation can take care of it and we don't need the `isinstance` or the `str(group_id)` anymore. That would be the best. We need to ensure two things:

1) that a UUIDField can be serialized by the celery JSON serializer (for an upcoming change)
2) that postgres supports UUID fields natively for when we switch to that